test/OrleansTestKit.Tests/Tests/TimerTests.cs

```diff
@@ -1,11 +1,59 @@
 ﻿using FluentAssertions;
+using Orleans.TestKit.Timers;
 using TestGrains;
 using Xunit;
 
 namespace Orleans.TestKit.Tests;
 
 public class TimerTests : TestKitBase
 {
+    [Fact]
+    public async Task ShouldFireFirstGrainTimerAsync()
+    {
+        // Arrange
+        var grain = await Silo.CreateGrainAsync<HelloTimers>(0);
+
+        // Act
+        await Silo.FireTimerAsync(HelloTimers.GrainTimer0);
+
+        // Assert
+        var state = Silo.State<HelloTimers, HelloTimersState>();
+        state.GrainTimer0Fired.Should().BeTrue();
+    }
+
+    [Fact]
+    public async Task ShouldFirstGrainTimerAsync()
```

Comment:
Is this an improved version of `ShouldFireFirstGrainTimerAsync` above? Do we want both?

Reply:
Not sure... this only tests the new RegisterGrainTimer timer, and the other tests the legacy timer. If you want them separate, we keep it.

Reply:
Oh, ther's a dupe -- I didn't notice. Let me check what I was thinking...

Reply:
Good catch -- I think I wrote that test before writing the cancellation bits, then added another test without thinking. I deleted the first one.